data/storage/bolt.go

```
@@ -942,41 +957,40 @@ func (self *BoltStorage) GetTokenTargetQty() (metric.TokenTargetQty, error) {
 	return tokenTargetQty, err
 }
 
+//StoreTokenTargetQty store token target quantity in db
 func (self *BoltStorage) StoreTokenTargetQty(id, data string) error {
 	var err error
 	var tokenTargetQty metric.TokenTargetQty
-	var dataJson []byte
+	var dataJSON []byte
 	err = self.db.Update(func(tx *bolt.Tx) error {
 		pending := tx.Bucket([]byte(PENDING_TARGET_QUANTITY))
 		_, pendingTargetQty := pending.Cursor().Last()
 
 		if pendingTargetQty == nil {
-			err = errors.New("There is no pending target activity to confirm.")
+			return errors.New("there is no pending target activity to confirm")
+		}
+		// verify confirm data
+		if err := json.Unmarshal(pendingTargetQty, &tokenTargetQty); err != nil {
 			return err
-		} else {
-			// verify confirm data
-			json.Unmarshal(pendingTargetQty, &tokenTargetQty)
-			pendingData := tokenTargetQty.Data
-			idInt, _ := strconv.ParseUint(id, 10, 64)
-			if tokenTargetQty.ID != idInt {
-				err = errors.New("Pending target quantity ID does not match")
-				return err
-			}
-			if data != pendingData {
-				err = errors.New("Pending target quantity data does not match")
-				return err
-			}
+		}
+		pendingData := tokenTargetQty.Data
+		idInt, _ := strconv.ParseUint(id, 10, 64)
```

Don't swallow conversion error